components/eam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -984,38 +986,41 @@ subroutine micro_p3_tend(state, ptend, dtime, pbuf)
     !---------------------------------------------------------------------------------------
-    qv_wet_in   = state%q(:,:,1) ! Get "wet" water vapor mixing ratio from state
     !Compute dry mixing ratios for all the constituents
-    qv_dry(:ncol,:pver)      = calculate_drymmr_from_wetmmr(ncol, pver, qv_wet_in,              qv_wet_in)
-    qv_prev_dry(:ncol,:pver) = calculate_drymmr_from_wetmmr(ncol, pver, qv_prev_wet,            qv_wet_in)
-    cldliq(:ncol,:pver)      = calculate_drymmr_from_wetmmr(ncol, pver, state%q(:,:,ixcldliq),  qv_wet_in)
-    numliq(:ncol,:pver)      = calculate_drymmr_from_wetmmr(ncol, pver, state%q(:,:,ixnumliq),  qv_wet_in)
-    rain(:ncol,:pver)        = calculate_drymmr_from_wetmmr(ncol, pver, state%q(:,:,ixrain),    qv_wet_in)
-    numrain(:ncol,:pver)     = calculate_drymmr_from_wetmmr(ncol, pver, state%q(:,:,ixnumrain), qv_wet_in)
-    ice(:ncol,:pver)         = calculate_drymmr_from_wetmmr(ncol, pver, state%q(:,:,ixcldice),  qv_wet_in)
-    qm(:ncol,:pver)          = calculate_drymmr_from_wetmmr(ncol, pver, state%q(:,:,ixcldrim),  qv_wet_in) !Aaron, changed ixqm to ixcldrim to match Kai's code
-    numice(:ncol,:pver)      = calculate_drymmr_from_wetmmr(ncol, pver, state%q(:,:,ixnumice),  qv_wet_in)
-    rimvol(:ncol,:pver)      = calculate_drymmr_from_wetmmr(ncol, pver, state%q(:,:,ixrimvol),  qv_wet_in)
+
+    ratio_local(:ncol,:pver) = state%pdel(:ncol,:pver)/state%pdeldry(:ncol,:pver)
+
+    qv_dry     (:ncol,:pver) = state%q(:ncol,:pver,         1) *ratio_local(:ncol,:pver)
+    cldliq     (:ncol,:pver) = state%q(:ncol,:pver,  ixcldliq) *ratio_local(:ncol,:pver)
+    numliq     (:ncol,:pver) = state%q(:ncol,:pver,  ixnumliq) *ratio_local(:ncol,:pver)
+    rain       (:ncol,:pver) = state%q(:ncol,:pver,    ixrain) *ratio_local(:ncol,:pver)
+    numrain    (:ncol,:pver) = state%q(:ncol,:pver, ixnumrain) *ratio_local(:ncol,:pver)
+    ice        (:ncol,:pver) = state%q(:ncol,:pver,  ixcldice) *ratio_local(:ncol,:pver)
+    !Aaron, changed ixqm to ixcldrim to match Kai's code
+    qm         (:ncol,:pver) = state%q(:ncol,:pver,  ixcldrim) *ratio_local(:ncol,:pver)
+    numice     (:ncol,:pver) = state%q(:ncol,:pver,  ixnumice) *ratio_local(:ncol,:pver)
+    rimvol     (:ncol,:pver) = state%q(:ncol,:pver,  ixrimvol) *ratio_local(:ncol,:pver)
+
+    qv_prev_dry(:ncol,:pver) = qv_prev_wet(:ncol,:pver)        *ratio_local(:ncol,:pver)
 
     ! COMPUTE GEOMETRIC THICKNESS OF GRID & CONVERT T TO POTENTIAL TEMPERATURE
     !==============
     ! TODO: Create a general function to calculate Exner's formula that can be
     ! used by all parameterizations, such as P3 and SHOC.
     ! This would take a bit more work, so we have decided to delay this task
     ! until a later stage of code cleanup.
-    inv_exner(:ncol,:pver) = 1._rtype/((state%pmiddry(:ncol,:pver)*1.e-5_rtype)**(rair*inv_cp))
+
+    inv_exner(:ncol,:pver) = 1._rtype/((state%pmid(:ncol,:pver)*1.e-5_rtype)**(rair*inv_cp))
```

Comment:
Just to clarify, is this change being made because it matters for conservation, or is it just because using dry mass for the exner function doesn't make much sense in the first place. I think this change looks good, just want to understand.

Reply:
This does not affect conservation, but yes, "dry" Exner does not make sense to me here.